create_pdfs.py

```diff
@@ -81,13 +85,31 @@ def main():
     certificate_data = []
 
     if args.course_id:
-        course_list = [args.course_id]
+        course_id = args.course_id
+        # Default course name to course_id when not supplied
+        # This is helpful when batch testing certs when the name is not
+        # readily available
+        long_course = args.long_course or course_id
+        course_list = [(course_id, args.assign_designation, long_course)]
     else:
-        course_list = settings.CERT_DATA.keys()
-
+        # Compile course list based on openedx-certificates/cert-data.yml
+        # gen_cert.py should and will fail if
+        # certificates-templates/cert-data.yml is used
+        course_list = []
+        for course_id, course_template in settings.CERT_DATA.iteritems():
+            course_list.append((course_id, None, course_id))
+            # Adding extra courses for each designation and subtemplate
+            designations = course_template.get('designations', {})
+            for key, value in designations.iteritems():
+                for title in value['titles']:
+                    course_list.append((course_id, title, course_id))
+            subtemplates = course_template.get('subtemplates', {})
+            for title in subtemplates.keys():
+                course_list.append((course_id, title, course_id))
     upload_files = not args.no_upload
 
-    for course in course_list:
+    cert = None  # Optimization for reusing previous template.
```

Comment:
Mentioned elsewhere, but this doesn't add to the conversation.
If this comment is removed, a developer isn't going to look at this line,
get confused, and wonder why we're declaring this variable.
It's clear from the code that we're declaring this variable so that it exists before we reference it.